tempest/src/test/kotlin/app/cash/tempest/CodecTest.kt

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.tempest
+
+import app.cash.tempest.musiclibrary.AlbumInfo
+import app.cash.tempest.musiclibrary.AlbumTrack
+import app.cash.tempest.musiclibrary.MusicDb
+import app.cash.tempest.musiclibrary.MusicDbTestModule
+import app.cash.tempest.musiclibrary.MusicItem
+import misk.aws.dynamodb.testing.DockerDynamoDb
+import misk.testing.MiskExternalDependency
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatIllegalArgumentException
+import org.junit.jupiter.api.Test
+import java.time.LocalDate
+import javax.inject.Inject
+
+@MiskTest(startService = true)
+class CodecTest {
+
+  @MiskTestModule
+  val module = MusicDbTestModule()
+
+  @MiskExternalDependency
+  val dockerDynamoDb = DockerDynamoDb
+
+  @Inject
+  lateinit var musicDb: MusicDb
+
+  @Test
+  internal fun itemCodecToDb() {
+    val albumInfoCodec = musicDb.music.codec(AlbumInfo::class)
```

Comment:
Could we provide an inline extension function?

```kotlin
inline fun <reified RI : Any, reified T: Any> LogicalTable<RI>.codec(): Codec<T, RI> {
  return codec(T::class)
}
```

Reply:
I tried it, but the callsite sucks:

```
    val albumInfoCodec = musicDb.music.codec(AlbumInfo::class)
```

goes to this:

```
    val albumInfoCodec = musicDb.music.codec<MusicItem, AlbumInfo>()
```

Any idea how to do this without having to repeat `MusicItem` ?

Reply:
I see. 👍 👍